### ISSUE TITLE
Doesn't work with ConfigurationProperty?

### ISSUE THREAD

**Issue by abhimanyuseth-kiva** [ericdallo/spring-s3-properties-loader#13]
I'm trying to load a configuration object from a YAML file in S3.

My YAML file is
```
test:
  s3property: "Success!!"
```

I have
```
@Configuration
@S3PropertiesLocation("testbucket/config/test.yaml")
class S3Config {
}
```

And
```
@Component
@ConfigurationProperties("test")
class TestConfig {
    var s3property: String = ""
}
```

When I Autowire TestConfig, it doesn't hold any value in s3property.

However, it works if I explicitly add @Value annotation to read the property.
```
@Component
@ConfigurationProperties("test")
class TestConfig {
    @Value("\${s3property}")
    var s3property: String = ""
}
```

Using @Value annotation is not practical since I have an object with map of lists.
I would like to be able to read the configuration into an object like we can with Spring application.yaml.

**Comment by ericdallo** [comment]
Yeah, this project just load the config and do not bind with any @ConfigurationProperties :/
sorry, but I also don't keep/use it since years, but PRs are very welcome :)

**Comment by abhimanyuseth-kiva** [comment]
Thanks! Would love to help here. Before I start researching on this, do you have any idea/thoughts what needs to be done to implement this?

**Comment by ericdallo** [comment]
Hum, Not sure, But when I implemented all this project I just red the spring docs about how spring beans works and it was not hard, debugging in eclipse helps too to understand the bean's lifecycle.
Maybe start reading how spring bind the @ConfigurationProperties values would help

**Comment by abhimanyuseth** [comment]
Thanks! I think I'm almost there. Can you grant me permission to push a branch and raise PR?
Username abhimanyuseth

**Comment by abhimanyuseth** [comment]
@ericdallo 
I have a branch ready with the following changes:
- Update library versions to run with the latest JDK, and Gradle version.
- Use YamlPropertiesFactoryBean to load properties when file is a YAML file
- Extend S3PropertySource from EnumerablePropertySource to allow binding to POJO Map properties.

**Comment by ericdallo** [comment]
Nice, done @abhimanyuseth !

**Comment by ericdallo** [comment]
Published release 3.0.0 supporting this, thank you!
